app/main.py

```diff
@@ -1,25 +1,78 @@
 from abc import ABC
+from typing import Any
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: Any, name: str) -> None:
+        self.public_name = name
+        self.protected_name = "_" + name
+
+    def __get__(self, instance: Any, objtype: Any = None) -> None:
+        value = getattr(instance, self.protected_name)
+        return value
+
+    def __set__(self, instance: Any, value: Any) -> None:
+        if self.validate(value):
+            setattr(instance, self.protected_name, value)
+        else:
+            setattr(instance, self.protected_name, None)
```

Comment:
You should raise ValueError in this case

Reply:
Fixed